### PR TITLE
initial commit for simple self-contained cuda example

### PR DESCRIPTION
**Pull request by anderslanglands**
## Description

This PR adds a simple, self-contained example as a test case called 'example-cuda'. It's basically just a stripped down version of testshade, wrapped up so that one could copy it out of tree and get it building as a standalone example with minimal changes to the CMakeLists.txt. I'm submitting this as I just started looking at OSL in CUDA recently and this is the example I would have loved to have started from.

This example is pure CUDA rather than OptiX, but I'd like to do a matching OptiX version at some point.

## Tests

It is a test :)

## Checklist:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have previously submitted a [Contributor License Agreement](http://opensource.imageworks.com/cla/).
- [ ] I have updated the documentation, if applicable.
- [x] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [x] My code follows the prevailing code style of this project.

### CODE REVIEW

File: testsuite/example-cuda/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# Is there a better way to get the location of the main OSL CMakeLists.txt?
+set(OSL_LIST_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../../")
+# We have to make the path absolute or CMake will fail to find the file...
+get_filename_component(OSL_LIST_DIR ${OSL_LIST_DIR} ABSOLUTE)
+set(CMAKE_MODULE_PATH "${OSL_LIST_DIR}/src/cmake;${OSL_LIST_DIR}/src/cmake/modules")
+
+include(check_is_enabled)
+include(checked_find_package)
+
+find_package(OSL REQUIRED)
```
Comment (lgritz) [inline testsuite/example-cuda/CMakeLists.txt:21]: You can use checked_find_package for all of these

**Review by lgritz — APPROVED** [review]
LGTM. This is a great addition, Anders.

**Comment by lgritz** [comment]
I'm ready to merge whenever you let me know you've worked it all out with your company and they are clear with you that this work is covered by the CLA they've already signed.

**Comment by anderslanglands** [comment]
All is good with the overlords but I'll need to give you an individual CLA.
Where should I email it to?

**Comment by lgritz** [comment]
For OSL, email CLAs to opensource@imageworks.com, for OIIO mail to lg@openimageio.org.

**Comment by lgritz** [comment]
CLA received.

Anders, can you please rebase this on top of the current master and then do a push --force? I think the CI failures are all things unrelated to your change, which have already been fixed. But I'd love to see a fully passed CI so we know there's not a problem lurking in this patch being masked by the other failures.

**Comment by anderslanglands** [comment]
Done and done. Unfortunately the Appveyor build appears to be failing on
the pugixml build?

**Comment by lgritz** [comment]
Don't worry about the appveyor. I should just kill that entirely.

**Pull request merged**